src/transformers/pipelines.py

```diff
@@ -1750,7 +1765,32 @@ def pipeline(
     """
     # Retrieve the task
     if task not in SUPPORTED_TASKS:
-        raise KeyError("Unknown task {}, available tasks are {}".format(task, list(SUPPORTED_TASKS.keys())))
+        if "translation" in task:
+            # Check that translation has special string format
+            # Using "translation_en_to_fr" should slowly be deprecated
+            # Instead "translation" should be used with "src_lang" and "tgt_lang" kwargs
+            if not ("src_lang" in kwargs and "tgt_lang" in kwargs):
+                warnings.warn(
+                    "When using translation pipeline, `src_lang` and `tgt_lang` should be provided in kwargs",
+                    FutureWarning,
+                )
+                assert "_to_" in task, "f{task} should be in the format 'translation_{src_lang}_to_{tgt_lang}"
+
+                langs = task.split("translation_")[-1]
+                src_lang, tgt_lang = langs.split("_to_")
+                kwargs["src_lang"] = src_lang
+                kwargs["tgt_lang"] = tgt_lang
+
+            assert (
+                isinstance(kwargs["src_lang"], str) and len(kwargs["src_lang"]) > 0
+            ), f"{kwargs['src_lang']} cannot be empty"
+            assert (
+                isinstance(kwargs["tgt_lang"], str) and len(kwargs["tgt_lang"]) > 0
+            ), f"{kwargs['tgt_lang']} cannot be empty"
+
+            task = "translation"
+        else:
+            raise KeyError("Unknown task {}, available tasks are {}".format(task, list(SUPPORTED_TASKS.keys())))
```

Comment:
crazy out of scope idea: if the language pair is `not in SUPPORTED_TASKS`, try to find a marian model to do it.

```python
marian_path = f'Helsinki-NLP/opus-mt-{src_lang}-{tgt_lang}'
try:
   tokenizer = AutoTokenizer.from_pretrained(marian_path)
   model = AutoModelForSeq2SeqLM.from_pretrained(marian_path)
except OSError:
   raise KeyError(f'Could not find model for {task} after checking {SUPPORTED_TASKS.keys()} and {marian_path}.')
```

Reply:
In terms of whether this breaks marian pipeline usage, IDT so, there is a test: https://github.com/huggingface/transformers/blob/master/tests/test_modeling_marian.py#L235